### PR TITLE
Geom: Integrate new `Ray` class

### PR DESCRIPTION
**Pull request by jsn0716**
### Description
Replace the old `Ray` class with the new one, the major difference between them being that the new one uses `Angle` instead of `Vector` to internally represent direction. Also moved the `inline` methods to a separate source file.

### Testing Done
Added a couple unit tests to `Ray`, and regression test

### Resolved Issues
#1096 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/software/new_geom/ray.h [hunk 1]
```diff
@@ -6,91 +6,74 @@ class Ray final
...
     /**
      * Rotates this Ray counterclockwise by an angle
      *
      * @param angle the angle to rotate the Ray
      */
-    inline void rotate(Angle angle)
-    {
-        direction_ += angle;
-    }
+    void rotate(Angle angle);
```
Comment (scveloso) [inline src/software/new_geom/ray.h]:
```suggestion
    void rotate(const Angle& angle);
```

File: src/software/new_geom/ray.cpp [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+{
+    direction_ = direction;
+}
+
+void Ray::setDirection(const Vector& direction)
+{
+    direction_ = direction.orientation();
+}
+
+void Ray::rotate(Angle angle)
```
Comment (scveloso) [inline src/software/new_geom/ray.cpp]:
```suggestion
void Ray::rotate(const Angle& angle)
```

**Review by scveloso — CHANGES_REQUESTED** [review]
LGTM aside from one small thing

**Review by MathewMacDougall — COMMENTED** [review]
LGTM pending Sam's comments

File: src/software/new_geom/ray.h [hunk 3]
```diff
@@ -6,91 +6,74 @@ class Ray final
 {
    public:
     /**
-     * Creates a degenerate Ray at (0, 0)
+     * Creates a Ray starting at (0, 0) with zero angle
```
Comment (garethellis0) [inline src/software/new_geom/ray.h]: Could we change this to 
``` suggestion
     * Creates a Ray starting at (0, 0) with a degenerate direction vector (ie. (0,0))
```
Also we aren't actually doing this in the default constructor? We should explicitly set these values to zero there.

Comment (jsn0716) [reply src/software/new_geom/ray.h]: Done

File: src/software/new_geom/ray.h (hunk 1 above)
Comment (jsn0716) [reply src/software/new_geom/ray.h]: Good catch, fixed

File: src/software/new_geom/ray.cpp (hunk 2 above)
Comment (jsn0716) [reply src/software/new_geom/ray.cpp]: Fixed

File: src/software/new_geom/ray.h (hunk 3 above)
Comment (garethellis0) [reply src/software/new_geom/ray.h]: Oh so we're storing the direction as an angle, so setting it to a direction of "zero" isn't degenerate right? We probably want to say something like "... at (0,0) pointing along the positive x-axis"

Comment (garethellis0) [reply src/software/new_geom/ray.h]: Sorry about that, misunderstood how we stored direction earlier.

**Review by scveloso — DISMISSED** [review]

File: src/software/new_geom/ray.h (hunk 3 above)
Comment (jsn0716) [reply src/software/new_geom/ray.h]: Fixed! Yeah direction of "zero" is just the zero angle.

**Comment by jsn0716** [comment]
@scveloso Fixed merge conflicts and a comment, please re-review!

**Review by MathewMacDougall — APPROVED** [review]

**Review by scveloso — APPROVED** [review]

**Pull request merged**